configuration.go

```diff
@@ -0,0 +1,156 @@
+package raft
+
+import "fmt"
+
+// ServerSuffrage determines whether a Server in a Configuration gets a vote.
+type ServerSuffrage int
+
+const (
+	// Voter is a server whose vote is counted in elections and whose match index
+	// is used in advancing the leader's commit index.
+	Voter ServerSuffrage = 1
```

Comment:
Would be slightly more idiomatic Go to use `iota` on this first one and leave the others blank to auto-number them.